pinot-common/src/main/java/org/apache/pinot/common/utils/request/RequestUtils.java

```diff
@@ -290,4 +303,90 @@ public static Map<String, String> getOptionsFromJson(JsonNode request, String op
   public static Map<String, String> getOptionsFromString(String optionStr) {
     return Splitter.on(';').omitEmptyStrings().trimResults().withKeyValueSeparator('=').split(optionStr);
   }
+
+  public static Set<String> getTableNames(SqlNode sqlNode) {
+    Set<String> tableNames = new HashSet<>();
+    if (sqlNode instanceof SqlSelect) {
+      // Handle SqlSelect query
+      SqlNode fromNode = ((SqlSelect) sqlNode).getFrom();
+      if ((fromNode instanceof SqlBasicCall)
+          && (((SqlBasicCall) fromNode).getOperator() instanceof SqlAsOperator)) {
+        tableNames.addAll(getTableNames(((SqlBasicCall) fromNode).getOperandList().get(0)));
+      } else if (fromNode instanceof SqlIdentifier) {
+        tableNames.add(getTableName((SqlIdentifier) fromNode));
+        tableNames.addAll(getTableNames(((SqlSelect) sqlNode).getWhere()));
```

Comment:
Can we skip `WHERE` clause for other if branches?
E.g. `SELECT * FROM myTable AS T1 WHERE col1 = (SELECT col2 FROM T2)`

Reply:
Need to check recursively.
See more examples in tests:
https://github.com/apache/pinot/pull/11330/files#diff-a4bccc2639d28b1ed4eeb105200ce6da89b1f53876909863134a7084c1aa2b8cR95
and 
https://github.com/apache/pinot/pull/11330/files#diff-a4bccc2639d28b1ed4eeb105200ce6da89b1f53876909863134a7084c1aa2b8cR147

Reply:
If I read correctly, the example query I posted will not check the `WHERE` clause because there is no recursion happening after line 314